python/hail/dataset.py

```diff
@@ -4582,6 +4582,72 @@ def delete_va_attribute(self, ann_path, attribute):
 
     @handle_py4j
     @requireTGenotype
+    @typecheck_method(key_name=strlike,
+                      variant_keys=strlike,
+                      single_key=bool,
+                      weight_expr=strlike,
+                      y=strlike,
+                      covariates=listof(strlike),
+                      hardCalls=bool)
+    def skat(self, key_name, variant_keys, single_key, weight_expr, y, covariates=[], hardCalls=True):
+        """Test each keyed group of variants for association by linear SKAT test.
+
+        .. include:: requireTGenotype.rst
+
+        **Examples**
+
+        Run a gene test using linear SKAT on hardCalls genotype data. Here ``va.genes``
+        is a variant annotation of type Set[String] giving the set of genes containing the variant:
+
+        # >>> skat_kt = (hc.read('data/example_skat.vds')
+        # ...     .skat(key_name='gene',
+        # ...           variant_keys='va.genes',
+        # ...           single_key=False,
+        # ...           weight='va.weight',
+        # ...           y='sa.skat.pheno',
+        # ...           covariates=['sa.cov1', 'sa.cov2']
+        # ...           hardCalls=True))
+
+        .. caution::
+
+          With ``single_key=False``, ``variant_keys`` expects a variant annotation of Set or Array type, in order to
+          allow each variant to have zero, one, or more keys (for example, the same variant may appear in multiple
+          genes). Unlike with type Set, if the same key appears twice in a variant annotation of type Array, then that
+          variant will be counted twice in that key's group. With ``single_key=True``, ``variant_keys`` expects a
+          variant annotation whose value is itself the key of interest. In both cases, variants with missing keys are
+          ignored.
+
+        **Notes**
+        
+        It should be noted that the variant component score (Q in the original SKAT paper) is returned scaled by 
+        \frac{1}{2\sigma^2}  where \sigma^2 is the unbiased variance computed in the least squares problem being 
+        solved. 
```

Comment:
you're right, I did mean to write variance component score, and the qstat is the Q value computed for each of the genes. I will also edit the table to reflect that results that I get for the 3 genes run from the burden test cases.